bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/AbstractArrayValue.java

```diff
@@ -309,6 +321,37 @@ public void setLength(long length) {
         size = newLength;
     }
 
+    protected void initializeIteratorNextReturnType() {
+        Map<String, BField> fields = new HashMap<>();
+        BType type;
+        if (getType().getTag() == TypeTags.ARRAY_TAG) {
+            type = getElementType();
+        } else {
+            BTupleType tupleType = (BTupleType) getType();
+            LinkedHashSet<BType> types = new LinkedHashSet<>(tupleType.getTupleTypes());
+            if (tupleType.getRestType() != null) {
+                types.add(tupleType.getRestType());
+            }
+            if (types.size() == 1) {
+                type = types.iterator().next();
+            } else {
+                type = new BUnionType(new ArrayList<>(types));
+            }
+        }
+
+        fields.put("value", new BField(type, "value", Flags.PUBLIC + Flags.REQUIRED));
+        iteratorNextReturnType = new BRecordType("$$returnType$$", null, 0, fields, null, true,
```

Comment:
Can't we use the constant for "$$returnType$$" here?

Reply:
fixed